airbyte-ci/connectors/pipelines/pipelines/models/steps.py

```diff
@@ -65,17 +65,25 @@ def is_file(self) -> bool:
         return self.get_path().is_file()
 
 
-@dataclass(frozen=True)
-class StepResult:
-    """A dataclass to capture the result of a step."""
-
-    step: Step
+@dataclass(kw_only=True)
```

Comment:
Because it's mutated in some tests... I can try to patch it at test time but was not sure how important it is to keep it frozen. 
I'll try to keep it frozen for sanity.

Reply:
It definitely feels safer to keep frozen unless needed. Why are the objects mutated in the tests if they're not mutated in the production code paths?

Reply:
@girarda I figured that the `__post_init__` was the reason a mutation happened.
With the `__post_init__` hook we reset the `stdout` and `stderr` attribute to a new value to redact secrets from these attributes. 
Setting these attributes in a different way avoids the error and allow us to keep frozen classes:
```python

    def __post_init__(self) -> None:
        if self.stderr:
            object.__setattr__(self, "stderr", self.redact_secrets_from_string(self.stderr))
        if self.stdout:
            object.__setattr__(self, "stdout", self.redact_secrets_from_string(self.stdout))
```